### PR TITLE
Questionnaire response sync resource added

### PR DESCRIPTION
**Pull request by siddh1004**
https://app.shortcut.com/simpledotorg/story/9558/add-sync-resource-for-questionnaire-responseapi

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/questionnaireresponse/sync/QuestionnaireResponsePullResponse.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.simple.clinic.sync.DataPullResponse
+
+@JsonClass(generateAdapter = true)
+data class QuestionnaireResponsePullResponse(
+
+    @Json(name = "questionnaire_responses")
+    override val payloads: List<QuestionnaireResponsePayload>,
+
+    @Json(name = "process_token")
+    override val processToken: String
```
Comment (olttwa) [inline app/src/main/java/org/simple/clinic/questionnaireresponse/sync/QuestionnaireResponsePullResponse.kt:14]: In the `adding-sync-resource.md` guide, there's a Step 3. which says:
> Save the `processToken` in a shared preference and provide it through the Dagger DI graph.

Are we doing this? We couldn't find it.

File: app/src/androidTest/java/org/simple/clinic/sync/QuestionnaireResponseSyncIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    }
+    Truth.assertThat(records).containsNoDuplicates()
+
+    questionnaireResponseRepository.save(records)
+    Truth.assertThat(questionnaireResponseRepository.pendingSyncRecordCount().blockingFirst()).isEqualTo(totalNumberOfRecords)
+
+    // when
+    questionnaireResponseSync.push()
+    clearQuestionnaireResponseData()
+    questionnaireResponseSync.pull()
```
Comment (olttwa) [inline app/src/androidTest/java/org/simple/clinic/sync/QuestionnaireResponseSyncIntegrationTest.kt:154]: Does pull change the SyncStatus of a record, or on successful push, the status gets marked as `DONE`?

**Review by olttwa — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/simple/clinic/questionnaireresponse/sync/QuestionnaireResponsePullResponse.kt (hunk 1 above)
Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/questionnaireresponse/sync/QuestionnaireResponsePullResponse.kt:14]: @olttwa yes we are doing this.The code can be found at `QuestionnaireResponseModule.kt`

File: app/src/androidTest/java/org/simple/clinic/sync/QuestionnaireResponseSyncIntegrationTest.kt (hunk 2 above)
Comment (siddh1004) [reply app/src/androidTest/java/org/simple/clinic/sync/QuestionnaireResponseSyncIntegrationTest.kt:154]: no, pull doesn't change the sync status. It's only updated on a push.

File: app/src/androidTest/java/org/simple/clinic/sync/QuestionnaireResponseSyncIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+import org.simple.clinic.rules.ServerAuthenticationRule
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.unsafeLazy
+import org.simple.sharedTestCode.TestData
+import org.simple.sharedTestCode.util.Rules
+import java.util.Optional
+import java.util.UUID
+import javax.inject.Inject
+
+@Ignore("the qa api is under development")
```
[the remark this answers is not in the document]
Comment (siddh1004) [reply app/src/androidTest/java/org/simple/clinic/sync/QuestionnaireResponseSyncIntegrationTest.kt:34]: @olttwa The api is still at a building phase, and the questionnaireSync and questionnaireResponseSync are still not part of sync module. When that happens, we will remove the ignore now thing.

**Review by olttwa — APPROVED** [review]

**Review by olttwa — APPROVED** [review]

**Pull request merged**